### PR TITLE
New WhatsApp BSP channel: 360dialog

### PR DESCRIPTION
**Pull request by koallann**
Submitting the PR https://github.com/rapidpro/rapidpro/pull/1280 in the correct repo this time.

### CODE REVIEW

**Review by nicpottier — APPROVED** [review]
Looks good.

**Comment by nicpottier** [comment]
Can you merge master in again and we'll merge?

**Comment by koallann** [comment]
done

**Comment by matmsa27** [comment]
Hi @nicpottier , do you have news about this PR? We running this in our environment for about 2 months and works fine.

**Comment by rowanseymour** [comment]
@koallann sorry but can you merge in master again and I promise we'll try to merge this before we create more conflicts again!

File: templates/channels/types/dialog360/config.haml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-load i18n
+
+%h4
+  -blocktrans
```
Comment (rowanseymour) [inline templates/channels/types/dialog360/config.haml:4]: should use `trimmed`

File: temba/channels/types/dialog360/views.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            return self.cleaned_data
+
+    form_class = Form
+
+    def form_valid(self, form):
+        user = self.request.user
+        org = user.get_org()
+
+        if not org:  # pragma: no cover
+            raise Exception(_("No org for this user, cannot claim"))
```
Comment (rowanseymour) [inline temba/channels/types/dialog360/views.py:41]: I guess you copied this from another channel type but it's not necessary and shouldn't be a localized string if it was any way. Can just remove.

File: temba/channels/types/dialog360/views.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        config = {
+            Channel.CONFIG_BASE_URL: data["base_url"],
+            Channel.CONFIG_AUTH_TOKEN: data["api_key"],
+        }
+
+        self.object = Channel.create(
+            org,
+            user,
+            data["country"],
+            "D3",
```
Comment (rowanseymour) [inline temba/channels/types/dialog360/views.py:54]: can use `self.channel_type`

**Review by rowanseymour — CHANGES_REQUESTED** [review]

**Review by rowanseymour — APPROVED** [review]

**Comment by rowanseymour** [comment]
Nevermind above comments, I'll cleanup with all the other channel types in https://github.com/nyaruka/rapidpro/pull/3050

**Pull request merged**